src/views/Simulation/components/Scene/Scene.js

```diff
@@ -80,8 +80,10 @@ const Scene = () => {
 
   useEffect(() => {
     if (!needsReRendering) return;
-    if (requiredUpdateStepsRef.current.all || requiredUpdateStepsRef.current.layout) {
+    const layoutUpdate = requiredUpdateStepsRef.current.all || requiredUpdateStepsRef.current.layout;
+    if (layoutUpdate) {
       resetGraphLayout(sceneCanvasRef.current.clientWidth, sceneCanvasRef.current.clientHeight);
+      setSelectedElementId(null);
```

Comment:
This change means that the selected element will be reset every time the layout is changed? Can't we keep the selected element when we switch between horizontal and vertical layouts?